Update README.md (#22)
etclabscore/ECIPs: README.md

When you believe your ECIP is mature and ready to progress past the draft phase, you should do one of two things:

 - **For a Standards Track ECIP of type Core**, ask to have your issue added to the agenda of an upcoming All Dev Team meeting, where it can be discussed for inclusion in a future hard fork. If implementers agree to include it, the ECIP editors will update the state of your ECIP to 'Accepted'.

>  - **For all other ECIPs**, open a PR changing the state of your ECIP to 'Final'. An editor will review your draft and ask if anyone objects to it being finalized. If the editor decides there is no rough consensus - for instance, because contributors point out significant issues with the ECIP - they may close the PR and request that you fix the issues in the draft before trying again.

Reviewer: ```suggestion
 - **For all other ECIPs**, open a PR changing the state of your ECIP to 'Final'. An editor will review your draft and ask if anyone objects to its finalization. If the editor decides there is no rough consensus - for instance, because contributors point out significant issues with the ECIP - they may close the PR and request that you fix the issues in the draft before trying again.
```

# ECIP Status Terms

> * **Draft** - An ECIP was submitted by its author and is undergoing rapid iteration and changes.

Reviewer 2: I don't know why this change is required. Technically, a submitted EIP only becomes a draft after being merged. Before it, it's just _in progress_.

Author: Thanks for clarifying. Returning to original text.

Author: Thanks for clarifying. Returning to original text.

* **Last Call** - An ECIP that is done with its initial iteration and ready for review by a wider audience.

> * **Accepted** - An ECIP that has been in Last Call for at least 2 weeks and any technical changes that were requested have been addressed by the author. If Core Devs decide to implement an ECIP into their clients, the ECIP will move to final.

Reviewer 2: That added sentence duplicates logic from _final_.

Author: It is a line that serves as a transition to the next status. In the current text, several status descriptions seemed the same and not very clear. As long as it is a duplication and serves as clarification, but is not technically wrong, I would leave it.

Author: Now that we simplified to just "Final" below, this line actually totally looks redundant so it deleted is.

> * **Final (non-Core)** - An ECIP that has been in Last Call for at least 2 weeks, any technical changes that were requested have been addressed by the author, and Core Devs decided to implement it into their clients. 

Reviewer 2: That reads as if a final ECIP can not have any implementation, which is technically incorrect. Final should always mark ECIPs that are beyond accepted, implemented, and in a state where it can no longer be modified.

Author: For some reason both Ethereum EIP and Ethereum Classic ECIP Status Terms have divided the "Final" status into two" Final (non-core) and Final (Core). 

It seems there is and intermediate period between "Accepted" and "Final"  where devs implement changes, but are not scheduled or deployed in a hard fork.

If that intermediate period is true or really exists, then we can change the term Final (non-Core) to "Intermediate" or "Pre-final".

If not we can strike "Final (non-Core)" altogether.

Please let me know if that intermediate state actually exists or what terminology you would use.

Reviewer 2: Yeah, I think we should simplify it here.

Draft > Last Call > Accepted > Final; I don't see how non-core proposals should follow a different route.

Author: Done.

Also I unified both "final" sections and used the text you suggested below.

> * **Final (Core)** - An ECIP that the Core Devs have decided to implement and release in a future hard fork or has already been released in a hard fork. 

Reviewer 2: This is also not clean, core proposals can only be final once activated on a network, so after a hardfork on testnet or mainnet.

Author: Agreed. Would this sentence work?

**Final (Core)** - An ECIP that the Core Devs have decided to implement and has been deployed in a hard fork.

Reviewer 2: ```suggestion
* **Final** - An ECIP that was accepted, implemented, and no longer can be modified without submitting a new proposal, e.g., it has been released in a hard fork. 
```

* **Deferred** - An ECIP that is not being considered for immediate adoption. May be reconsidered in the future for a subsequent hard fork.
* **Replaced** - When a Final ECIP is no longer relevant, its status may be changed to Replaced or Obsolete.

> * **Rejected** - Reasons for rejecting ECIPs include duplication of effort, disregard for formatting rules, unfocused or too broad, being technically unsound, not providing proper motivation, or conflicting with the decentralization, backward compatibility, and state finality of the network.

Reviewer: ```suggestion
* **Rejected** - Reasons for rejecting ECIPs include duplication of effort, disregard for formatting rules, unfocused or too broad, being technically unsound, not providing proper motivation, or obvious popular disapproval.
```

>  - **For all other ECIPs**, open a PR changing the state of your ECIP to 'Final'. An editor will review your draft and ask if anyone objects to its being finalised. If the editor decides there is no rough consensus - for instance, because contributors point out significant issues with the ECIP - they may close the PR and request that you fix the issues in the draft before trying again.

Reviewer 3: ```suggestion
 - **For all other ECIPs**, open a PR changing the state of your ECIP to 'Final'. An editor will review your draft and ask if anyone objects to it being finalized. If the editor decides there is no rough consensus - for instance, because contributors point out significant issues with the ECIP - they may close the PR and request that you fix the issues in the draft before trying again.
```

# ECIP Status Terms
* **Draft** - An ECIP was submitted by its author and is undergoing rapid iteration and changes.
* **Last Call** - An ECIP that is done with its initial iteration and ready for review by a wider audience.
* **Accepted** - An ECIP that has been in Last Call for at least 2 weeks and any technical changes that were requested have been addressed by the author. If Core Devs decide to encode an ECIP into their clients, the ECIP will move to final.
* **Final (non-Core)** - An ECIP that has been in Last Call for at least 2 weeks, any technical changes that were requested have been addressed by the author, and Core Devs decided to encode it into their clients. 
* **Final (Core)** - An ECIP that the Core Devs have decided to implement and release in a future hard fork or has already been released in a hard fork. 
* **Deferred** - An ECIP that is not being considered for immediate adoption. May be reconsidered in the future for a subsequent hard fork.

> * **Replaced/Obsolete** - When a Final ECIP is no longer relevant, its status may be changed to Replaced or Obsolete (which is equivalent to Replaced).

Reviewer 3: Obsolete ECIP can lead to confusing since it's already related to Replaced. Imagine reintroducing Obsolete ECIPs. To omit and ecip requires an ecip. The content of the ECIP already makes sense of obsoleting anything.

Author: The text is from ECIP-1000, but I agree, let's leave only replaced which implies 'obsolete' as well.

> * **Rejected** - Reasons for rejecting ECIPs include duplication of effort, disregard for formatting rules, being too unfocused or too broad, being technically unsound, not providing proper motivation or addressing backwards compatibility, or not in keeping with the Ethereum Classic philosophy.

Reviewer 3: ```suggestion
* **Rejected** - Reasons for rejecting ECIPs include duplication of effort, disregard for formatting rules, unfocused or too broad, being technically unsound, not providing proper motivation, or conflicting with the decentralization, backward compatibility, and state finality of the network.
```
"ETC philosophy" doesn't describe any specification unless defined. Might as well list the design principles.

Author: The text is from ECIP-1000 which is meant to be a specification, but i agree it's broad and vague. Might as well leave it out because listing the principles would make it too long.

> * **Withdrawn** - ECIP authors may decide on their own to change the status between Draft, Deferred, or Withdrawn. The ECIP editor may also change the status to Deferred when no progress is being made on the ECIP.

Reviewer 3: ```suggestion
* **Withdrawn** - ECIP authors may decide to change the status between Draft, Deferred, or Withdrawn. The ECIP editor may also change the status to Deferred if no progress is being made on the ECIP.
```
More concise suggestions.

Author: Perfect.
